viite-UI/src/view/ProjectEditForm.js

```diff
@@ -467,8 +476,17 @@
         setFormDirty();
       });
 
-      rootElement.on('change', '.form-select-control', function () {
-        setFormDirty();
+        rootElement.on('change', '.form-select-control', function () {
+            setFormDirty();
+        });
+
+      rootElement.on('change', '#tie', function () {
+        const tie = $('#tie');
+        if (tie.val() > 20000 && tie.val() < 40000) {
+           $('#osa').val(1);
+           $('#trackCodeDropdown').val("0");
+           $('#discontinuityDropdown').val("1");
+       }
```

Comment:
miksi? tienumerolla väliltä ]20000, 40000[ voi olla tieosa > 1, mutta kyseessä ei ole silloin kiertoliittymä. Omaan silmään tämä näyttää tarpeettomalta.

Reply:
Siis jos nyt ymmärrän oikein, ja tää asettaa tieosanumerokai aina 1 kun tien numero on 20k-40k, niin sitten tuo on substanssivirhe. On muitakin kuin 1-tieosia tossa 20k-40k-haarukassa.
Jos on tarkoitus säätää tietoja kun kyseessä on kiertoliittymä, sit ton _$('#osa').val(1);_ sijaan _osa.val==1_ iffin sisälle.
Yksiajorataisia kiertoliittymät luullakseni ovat aina (kaistoja voi olla useita).
Mutta sit taas toi jatkuvuus... Missäs kohtaa tää ns. kicks in... Ei kai jatkuvuuttakaan voi aina tien lopuksi säätää; voinee tulla tilanteita, joissa sitä kiertoliittymää ei voi osoitteistaa kerralla (vaikka lienevät harvinaisia)..?